src/subscriptions.c

```diff
@@ -952,6 +954,24 @@ subscription_done(void)
  * *************************************************************************/
 
 /**
+ * Update incoming byte count
+ */
+void subscription_add_bytes_in(th_subscription_t *s, size_t in)
+{
+  atomic_add(&s->ths_bytes_in, in);
+  atomic_add_u64(&s->ths_total_bytes_in, in);
```

Comment:
`ths_bytes_in_prev` is/should only be updated by the callback since it has to be updated only once per second. I guess I just still don't understand your reasoning here. IMO it doesn't really matter if the total byte count differs slightly.